OpenRCT2/OpenRCT2
# Move all libs to dependencies builds

![alt text](http://api.ning.com/files/kV4MbYiv7oR-EZ9ikEUgSva3KX14KAq3WLJe2FnU1965eWD7ANDyN1kiqx6xli3VxQHzInmYzgDggX5CgQN4kvMvjlRKTsCP/1082083033.jpeg "Library")

Well, it's finally been done. It took over a week and a bunch of testing, but lib is now completely removed from the repo and the prebuild now fetches the platform relevant dependencies from the openrct server.

What does this mean?

1. When the PR passes through, please all delete your lib folder, so that the prebuild script can download the new lib built from the dependencies repo.
2. No more adding libs to the OpenRCT2/OpenRCT2. If you have a new dependency that you would like to be added, you have to add it to the OpenRCT2/Dependencies and update the build scripts in that repo, and submit a PR for @IntelOrca or myself to review and test. Once that is gone through the lib will be included in the official dependencies zip.
3. We still don't have a good way of updating libs on the developer machines. We don't think the libs will be built that often so this probably won't be that much of an issue, but basically for now if the libs update, you have to delete the lib folder if you are on windows and using vs or run clean.sh and install.sh if you are on unix/building with mingw.

Before this is merged, I would really love it if a few people tested it out.
To test it here is what you have to do:
* On Windows using VS
 * delete the lib folder
 * build the solution
 * if everything is ok, that should be it, just test out if the game is ok
* On Unix/using MinGW
 * ./clean.sh
 * ./install.sh
 * ./build.sh && wine opnerct2.exe
 * if everything is ok, that should be it, just test out if the game is ok

If you would rather download the libs manually they are available on links bellow
* [VS version](https://download.openrct2.website/dev/lib/vs)
* [MinGW version](https://download.openrct2.website/dev/lib/mingw)

The MinGW version has a bit of a silly layout, but if you look into install.sh it will make sense.
